### DISCUSSION TITLE
What test data updates will we need in {pharmaversesdtm} for {admiralpeds}? See https://github.com/pharmaverse/pharmaversesdtm/issues/88

### DISCUSSION THREAD

**Discussion by rossfarrugia** [pharmaverse/admiralpeds#4] [General]
Example: ensuring subjects in DM exist with different pediatric ages.

**Comment by bms63** [comment]
maybe a separate data package should be created?  

Can also just make data in the package and move out once things are more stable.

**Reply by rossfarrugia** [reply]
Any thoughts on this @manciniedoardo @kaz462? i thought we were trying to keep all the test data together in {pharmaversesdtm}?

**Reply by manciniedoardo** [reply]
This is the first TA extension package after the establishment of {pharmaversesdtm}. As such, for all past TA extensions we had already developed the test datasets within the {admiral<ta_extension>} package and so just moved them over to the SDTM package. But now that we already have {pharmaversesdtm} I would recommend creating the test datsets directly within {pharmaversesdtm}. If you need variants of existing datasets, we can use the "_peds" extension  (eg `dm_peds`).

**Reply by bms63** [reply]
I think this will slow down development and create confusion if you have folks depend on another package to access data.  I really recommend to develop data in the package first and then move it out when it gets more stable.  

Just trying to minimize the steps for devs

**Reply by manciniedoardo** [reply]
Sure, you could develop in {admiralpeds} to start also, but I would sitll move everything prior to the first release.

**Reply by rossfarrugia** [reply]
It worked ok for admiralonco - we just did that step first back then in admiral.test. The test data only really drives the template/vignette and these are the last things we come to. for testing the functions we'll have our own unit tests so i think it could be fine but we can see what the dev team think.

**Reply by ngiangre** [reply]
Whichever solution allows for developing functions and tests faster, let's try that. If the data can be made available fast in {pharmaversesdtm}, then I'd go with putting data there. If not, add data to {admiralpeds} then move out. Probably the latter will be faster, is that a correct assumption?

**Reply by manciniedoardo** [reply]
@ngiangre you are correct, adding in `{admiralpeds}` is faster in the short term as you keep all development within one repo. However it creates slightly more work in the medium/long-term as you will have to move the data to `{pharmaversesdtm}` anyway prior to your first release. 

@zdz2101 feel free to make a call here in one direction or the other. At the end of the day both approaches will work, and the `{admiral}` team itself is split as to which path to pick, so we can leave to your discretion so that you select what works best for your pediatrics team. @kaz462 and I can help support from the `{pharmaversesdtm}` side when the time comes.

**Reply by zdz2101** [reply]
I think we can go ahead and proceed with an internal solution first then before release do the proper migration, but if anything, we can also look into Non-vignette articles where one of the use cases is: "Use of a package you don’t want to formally depend on. In vignettes and examples, it’s forbidden to show your package working with a package that you don’t list in DESCRIPTION, e.g. in Imports or Suggests."

**Reply by CDC-DNPAO** [reply]
I'd suggest using some of the data already on the CDC or WHO sites for test data.  For example, #8 at https://www.cdc.gov/growthcharts/percentile_data_files.htm 
gives various percentiles (starting with 'P') for BMI-for-age.  Similar percentiles are given for weight- and height-for-age.  I'd suggest using the 5th and 95th percentiles to test the CDC function.  For example, if you give the function a BMI of 19.2789 for a boy (sex=1) 24.5 months of age, the output should say that this child's BMI is at the 95th percentile.  I'd suggest using the 5th and 95th percentiles for test data. 

BMI values >95th percentile in the CDC growth charts are calculated differently from those below the 95th percentile.  For test data for these high BMI values, the BMII percentiles are at 
https://www.cdc.gov/growthcharts/extended-bmi-data-files.htm
If you click on the 'selected percentiles....', you'll get another excel file.  I suggest you focus on P99_99 (the 99.99th percentile) for testing.  One thing to note is that IMO, the guy who created this excel file didn't save enough digits after the decimal point, so the calculated percentiles and z-scores will be slightly different (~4th or 5th decimal place) that the output from cdcanthro.  You *might* also consider downloading the 'sigma' column (column F) so that you can reproduce the values on this web page exactly.

The WHO data are arranged somewhat differently, but if you go to 
https://www.who.int/tools/child-growth-standards/standards/weight-for-age 
and click on the expanded tables for boys' percentiles at the bottom, you'll see an excel file with various percentiles (P).  For WHO, I'd suggest using the P01 (the 0.1st percentile) 5th, 95, and P999  (the 99.9th percentile as test data).  WHO calculates z-scores outside -3 and +3 (the 0.14 and 99.86 percentiles) differently than those inside this range.

**Reply by CDC-DNPAO** [reply]
Oops - you might also consider downloading the 'z-score' expanded tables at the WHO website under the 'Expanded Tables'.  The give the weight, height, or BMI values at -4 or +4 z-scores (labelled SD4neg and SD4).  This would be a better test of extreme values in the whoanthro than the P01 and P999 percentiles I mentioned above.

**Comment by rossfarrugia** [comment]
Closing out this discussion now as we have actioned creating our initial test data, and we can incrementally improve these later as we see needed.
